Update binary-analysis for migration

Mihnea0Firoiu opened the pull request:
# Prerequisite Checklist

- [x] Read the [contribution guidelines](https://github.com/open-education-hub/ccas-internal/blob/main/CONTRIBUTING.md#pull-requests) regarding submitting new changes to the project;
- [x] Tested your changes against relevant architectures and platforms;
- [x] Updated relevant documentation (if needed).

## Description of changes

Update binary-analysis for migration.

teodutu commented on chapters/binary-introduction/binary-analysis/drills/tasks/easy-to-spot/README.md:
```diff
@@ -0,0 +1,5 @@
+
+It's an easy challenge.
+Really.
+
```
```suggestion
```

teodutu commented on chapters/binary-introduction/binary-analysis/drills/tasks/ghidra-killer/README.md:
```diff
@@ -0,0 +1,5 @@
+
+Some people just hate the people that use decompilers.
+One of those people left you a binary, to investigate.
+
```
```suggestion
```

teodutu commented on chapters/binary-introduction/binary-analysis/drills/tasks/packaging-is-important/README.md:
```diff
@@ -0,0 +1,4 @@
+# Packaging is Important
+
+Someone delivered you a mysterious package.
+
```
```suggestion
```

teodutu commented on chapters/binary-introduction/binary-analysis/drills/tasks/spaghetti/README.md:
```diff
@@ -0,0 +1,5 @@
+
+Someone felt like cooking today.
+Can you find the flag?
+
```
```suggestion
```

teodutu reviewed (changes requested):
- At the end of each task's `README.md`, add a sentence like, "If you're having difficulties solving this exercise, go through [this](link to the relevant section in reading/) reading material."
- Add the `src/` folder from each task to `support/` and leave the binary files at the same level as `src/`.

teodutu commented:
In addition, take a look at the failing CI/CD tests and fix them [1]. For the false positives coming from spellcheck, you'll have to add them here to the wordlists via a PR [2]. Use `sort -u -o <file-name> <file-name>` to keep those wordlists sorted.

[1] https://github.com/security-summer-school/essentials-security/pull/8/checks
[2] https://github.com/open-education-hub/actions/tree/main/spellcheck/config

Mihnea0Firoiu committed:
binary-analysis: Add session content

Add content related to binary-analysis.

Signed-off-by: Mihnea Firoiu <mihneafiroiu0@gmail.com>

teodutu commented on chapters/binary-introduction/binary-analysis/reading/README.md:
Remove the old "header":
```
---
linkTitle: 09. Binary Analysis
type: docs
weight: 10
---
```
And start the file with the first heading. Then split this file into 4 parts:

- `introduction.md` with the Reminders and Introduction
- `static-analysis.md`
- `dynamic-analysis.md`
- `summary.md`

Remember to update the link at the end of each task to point to the right reading material.

teodutu reviewed (changes requested):
Just split the reading material into 4 parts and then this should be good to go.

teodutu commented on chapters/binary-introduction/binary-analysis/reading/introduction.md:
To pacify super-linter, reverse the order of the sections in this file and turn "Reminders" into a level 2 heading (`##`).

teodutu reviewed (changes requested):
Just one more thing for super-linter to pass.

teodutu reviewed (changes requested):
I didn't see the C code files were also flagging super-linter [1]. First rebase the `main` branch because I added a config file for cpplint which will eliminate some of the errors. Then fix the remaining ones.

[1] https://github.com/security-summer-school/essentials-security/actions/runs/9974980826/job/27563650264?pr=8#step:3:625

Mihnea0Firoiu commented:
> I didn't see the C code files were also flagging super-linter [1]. First rebase the `main` branch because I added a config file for cpplint which will eliminate some of the errors. Then fix the remaining ones.

I rebased it, but that commit doesn't seem to change something regarding how cpplint works.

teodutu commented:
> > I didn't see the C code files were also flagging super-linter [1]. First rebase the `main` branch because I added a config file for cpplint which will eliminate some of the errors. Then fix the remaining ones.
> 
> I rebased it, but that commit doesn't seem to change something regarding how cpplint works.

Fuck, you,re right. I had only added `CPPLINT.cfg` to the internal repo. Now I've added it here as well. Rebase again and pull the latest `main` branch locally before doing so.
